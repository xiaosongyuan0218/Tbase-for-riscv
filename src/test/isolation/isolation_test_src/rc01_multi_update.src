#there are three records initialized
#diskey for node1/node2/node3 is 3/5/1 accordingly

setup
{
    create table tbl_xx (i int primary key, j int) distribute by shard(i);
}
setup
{
    insert into tbl_xx values (3, 1), (5, 2), (1, 3);
}

teardown
{
    DROP TABLE tbl_xx;
}

session "s1"
step "s1_begin"     { begin; }
step "s1_update"    { update tbl_xx set j = j+1; }
step "s1_select"    { select i,j from tbl_xx order by i; }
step "s1_commit"    { commit; }


session "s2"
step "s2_begin"     { begin; }
step "s2_select"    { select i,j from tbl_xx order by i; }
step "s2_commit"    { commit; }

permutation "s1_begin" "s1_select" "s2_select" "s1_update" "s1_select" "s2_select" "s1_commit" "s1_select" "s2_select"



