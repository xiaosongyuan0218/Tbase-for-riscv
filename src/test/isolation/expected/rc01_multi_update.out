Parsed test spec with 2 sessions

starting permutation: s1_begin s1_select s2_select s1_update s1_select s2_select s1_commit s1_select s2_select
step s1_begin: begin;
step s1_select: select i,j from tbl_xx order by i;
i              j              

1              3              
3              1              
5              2              
step s2_select: select i,j from tbl_xx order by i;
i              j              

1              3              
3              1              
5              2              
step s1_update: update tbl_xx set j = j+1;
step s1_select: select i,j from tbl_xx order by i;
i              j              

1              4              
3              2              
5              3              
step s2_select: select i,j from tbl_xx order by i;
i              j              

1              3              
3              1              
5              2              
step s1_commit: commit;
step s1_select: select i,j from tbl_xx order by i;
i              j              

1              4              
3              2              
5              3              
step s2_select: select i,j from tbl_xx order by i;
i              j              

1              4              
3              2              
5              3              
